creator/data_templates/mutations/data_template.py

```diff
@@ -0,0 +1,217 @@
+import graphene
+from graphql import GraphQLError
+from graphql_relay import from_global_id
+from django.conf import settings
+
+from creator.organizations.models import Organization
+from creator.data_templates.models import DataTemplate
+from creator.data_templates.nodes.data_template import DataTemplateNode
+
+
+class CreateDataTemplateInput(graphene.InputObjectType):
+    """Parameters used when creating a new data_template"""
+
+    name = graphene.String(
+        required=True, description="The name of the data_template"
+    )
+    description = graphene.String(
+        required=True, description="The description of the data_template"
+    )
+    icon = graphene.String(
+        description="Name of the Font Awesome icon to use when displaying the "
+        "template"
+    )
+    organization = graphene.ID(
+        required=True,
+        description="The organization that will own this data_template",
+    )
+
+
+class UpdateDataTemplateInput(graphene.InputObjectType):
+    """Parameters used when updating an existing data_template"""
+
+    name = graphene.String(description="The name of the data_template")
+    description = graphene.String(
+        description="The description of the data_template"
+    )
+    icon = graphene.String(
+        description="Name of the Font Awesome icon to use when displaying the "
+        "template"
+    )
+    organization = graphene.ID(
+        description="The organization that will own this data_template"
+    )
+
+
+class CreateDataTemplateMutation(graphene.Mutation):
+    """Creates a new data_template"""
+
+    class Arguments:
+        input = CreateDataTemplateInput(
+            required=True, description="Attributes for the new data_template"
+        )
+
+    data_template = graphene.Field(DataTemplateNode)
+
+    def mutate(self, info, input):
+        """
+        Creates a new data_template.
+        """
+        user = info.context.user
+        if not user.has_perm("data_templates.add_datatemplate"):
+            raise GraphQLError("Not allowed")
+
+        # Check if organization exists
+        org_id = input.pop("organization")
+        model, org_id = from_global_id(org_id)
```

Comment:
it might be too late because the db fields are set to a different type - but you can actually just use the global_ids as db ids. No need to call from_global_id/to_global_id all the time this way.

Reply:
I think we just use the relay id's for caching and lookup purposes bc we want some of our IDs to be "kids first style" (`SD_ME0WME0W` - crockford encoded with the 2 letter prefix)

Reply:
i've been under the impression the advantage of the crockford IDs in some instances is the 2 letter prefix allowing people to immediately know what type of data object they're looking at